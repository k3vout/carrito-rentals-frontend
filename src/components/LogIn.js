import React, { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { actions } from '../redux/app/app';
import Alert from './modules/Alert';
import splash from '../images/assets/carrito-splash.png';

const LogIn = () => {
  const alert = useSelector((state) => state.alertReducer);
  const dispatch = useDispatch();
  const [signUpOpen, setSignUpWindow] = useState(false);
  const [username, setUsername] = useState('');
  const [newusername, setNewUsername] = useState('');

  const handleSubmit = () => {
    if (signUpOpen) {
      const loginField = document.getElementById('login_field');
      loginField.className = 'align-self-center';
      setSignUpWindow(false);
    } else {
      dispatch(actions.logIn(username));
    }
  };
  const openSignUpPopUp = () => {
    if (signUpOpen) {
      dispatch(actions.signUp(newusername));
    } else {
      const loginField = document.getElementById('login_field');
      loginField.className = `${loginField.className} d-none`;
      setSignUpWindow(true);
    }
  };
  return (
    <div className="vh-100 text-center bg-warning d-flex flex-column justify-content-center">
      <div>
        <img
          src={splash}
          id="slide"
          className="carrito-splash-pic"
          alt="Carrito Logo"
        />
      </div>
      <div id="login_field" className="align-self-center">
        {alert ? <Alert message={alert} /> : false }
        <div className="input-group mb-5 username">
          <span className="input-group-text">@</span>
          <input type="text" placeholder="Username" onChange={(e) => setUsername(e.target.value)} className="form-control" />
        </div>
      </div>
      {signUpOpen ? (
        <div className="align-self-center">
          <p className="creators text-muted text-center">
            <small>
              Please create a new username:
            </small>
          </p>
          <div className="input-group mb-5 align-self-center username">
            <span className="input-group-text">New</span>
            <input type="text" placeholder="Username" onChange={(e) => setNewUsername(e.target.value)} className="form-control" />
          </div>
        </div>
      ) : false}
      <div className="input-group mb-5 align-self-center username text-center justify-content-center">
        <button type="button" onClick={() => { handleSubmit(); }} className="btn btn-dark btn-sm mx-1">Log in</button>
        <button type="button" onClick={() => { openSignUpPopUp(); }} className="btn btn-dark btn-sm mx-1">Sign Up</button>
      </div>
    </div>
  );
};

export default LogIn;
