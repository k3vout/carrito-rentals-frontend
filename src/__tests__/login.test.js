import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import LogIn from '../components/LogIn';
import store from '../redux/configureStore';

describe('Carrito Rental', () => {
  it('Renders login page:', () => {
    render(<Provider store={store}><LogIn /></Provider>);
    expect(screen.getByText('Log in')).toBeInTheDocument();
  });

  it('LogIn snapshot test', () => {
    const myrender = render(
      <Provider store={store}>
        <BrowserRouter>
          <LogIn />
        </BrowserRouter>
      </Provider>,
    );
    expect(myrender).toMatchSnapshot();
  });
});
