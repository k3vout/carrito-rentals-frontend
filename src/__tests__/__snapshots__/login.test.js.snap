// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carrito Rental LogIn snapshot test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="vh-100 text-center bg-warning d-flex flex-column justify-content-center"
      >
        <div>
          <img
            alt="Carrito Logo"
            class="carrito-splash-pic"
            id="slide"
            src="carrito-splash.png"
          />
        </div>
        <div
          class="align-self-center"
          id="login_field"
        >
          <div
            class="input-group mb-5 username"
          >
            <span
              class="input-group-text"
            >
              @
            </span>
            <input
              class="form-control"
              id="loginInput"
              placeholder="Username"
              type="text"
            />
          </div>
        </div>
        <div
          class="input-group mb-5 align-self-center username text-center justify-content-center"
        >
          <button
            class="btn btn-dark btn-sm mx-1"
            id="loginBtn"
            type="button"
          >
            Log in
          </button>
          <button
            class="btn btn-dark btn-sm mx-1"
            type="button"
          >
            Sign Up
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="vh-100 text-center bg-warning d-flex flex-column justify-content-center"
    >
      <div>
        <img
          alt="Carrito Logo"
          class="carrito-splash-pic"
          id="slide"
          src="carrito-splash.png"
        />
      </div>
      <div
        class="align-self-center"
        id="login_field"
      >
        <div
          class="input-group mb-5 username"
        >
          <span
            class="input-group-text"
          >
            @
          </span>
          <input
            class="form-control"
            id="loginInput"
            placeholder="Username"
            type="text"
          />
        </div>
      </div>
      <div
        class="input-group mb-5 align-self-center username text-center justify-content-center"
      >
        <button
          class="btn btn-dark btn-sm mx-1"
          id="loginBtn"
          type="button"
        >
          Log in
        </button>
        <button
          class="btn btn-dark btn-sm mx-1"
          type="button"
        >
          Sign Up
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
